packages/mui-base/src/PopperUnstyled/PopperUnstyled.tsx

```diff
@@ -33,7 +38,9 @@ function flipPlacement(placement, direction) {
   }
 }
 
-function resolveAnchorEl(anchorEl) {
+function resolveAnchorEl(
+  anchorEl: VirtualElement | (() => VirtualElement)
+): any {
```

Comment:
Not sure how to resolve this `any`. Seems `anchorEl` has the type `HTMLElement | VirtualElement | (() => HTMLElement | VirtualElement)` but the current code treat it as `HTMLElement | (() => HTMLElement)`

Reply:
while the original signature put it as `VirtualElement | (() => VirtualElement)`. Is there a potential bug here?

Reply:
I resolved this issue